sdk/communication/Azure.Communication.MediaComposition/src/MediaCompositionClient.cs

```diff
@@ -0,0 +1,320 @@
+        { }
+
+        private MediaCompositionClient(string endpoint, HttpPipeline httpPipeline, MediaCompositionClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RestClient = new MediaCompositionRestClient(_clientDiagnostics, httpPipeline, new Uri(endpoint), options.ApiVersion);
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="MediaCompositionClient"/> for mocking.</summary>
+        protected MediaCompositionClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<MediaCompositionBody>> GetAsync(string mediaCompositionId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Get");
+            scope.Start();
+            try
+            {
+                return await RestClient.GetAsync(mediaCompositionId, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response<MediaCompositionBody> Get(string mediaCompositionId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Get");
+            scope.Start();
+            try
+            {
+                return RestClient.Get(mediaCompositionId, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id. </param>
+        /// <param name="layout"> The layout to compose the media. </param>
+        /// <param name="inputs"> The media inputs. </param>
+        /// <param name="outputs"> The media outputs. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<MediaCompositionBody>> CreateAsync(string mediaCompositionId, MediaCompositionLayout layout, IDictionary<string, MediaInput> inputs, IDictionary<string, MediaOutput> outputs, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Create");
+            scope.Start();
+            try
+            {
+                // TODO: what is this id and is it needed?
```

Comment:
Commenting for visibility: What is this second Id and can we merge it with the Media composition Id?

Reply:
Had a discussion with team, we will be trying to remove the second `id` property because the two point to the same thing.

Reply:
Doesn't look like removing the id parameter worked. @DominikMe what are the changes we can make on the controller side to remove the id? Since we are just passing back mediaCompositionId for both

Reply:
Discussed with Dom offline, we will look into removing the additional id parameter, but okay to keep as is for now